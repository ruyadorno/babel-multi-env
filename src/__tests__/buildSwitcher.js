import path from "path";
import fs from "fs";
import promisify from "util.promisify";

const readFile = promisify(fs.readFile);

describe("buildSwitcher module", () => {
  const { buildAndOutputSwitcher } = require("../buildSwitcher");
  const multiVersions = ["8.7", "7.6", "6.0", "4.0", "0.12"];
  const outDir = path.join(__dirname, "../../tmp/buildSwitcher");

  describe("buildAndOutputSwitcher function", () => {
    it("should compile file", async () => {
      await buildAndOutputSwitcher(
        { multiVersions, outDir },
        {
          babelrc: false
        },
        "buildAndOutputSwitcher.js"
      );
      const content = await readFile(
        path.join(outDir, "buildAndOutputSwitcher.js"),
        "utf8"
      );
      expect(content).toMatchSnapshot();
    });
  });
});
