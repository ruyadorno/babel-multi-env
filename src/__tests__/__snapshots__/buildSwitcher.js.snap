// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildSwitcher module buildAndOutputSwitcher function should compile file 1`] = `
"var gte = require(\\"semver\\").gte;

var version = process.version;

if (gte(version, \\"8.7\\")) {
  module.exports = require(\\"./buildAndOutputSwitcher__8.7__.js\\");
} else if (gte(version, \\"7.6\\")) {
  module.exports = require(\\"./buildAndOutputSwitcher__7.6__.js\\");
} else if (gte(version, \\"6.0\\")) {
  module.exports = require(\\"./buildAndOutputSwitcher__6.0__.js\\");
} else if (gte(version, \\"4.0\\")) {
  module.exports = require(\\"./buildAndOutputSwitcher__4.0__.js\\");
} else {
  module.exports = require(\\"./buildAndOutputSwitcher__0.12__.js\\");
}"
`;
